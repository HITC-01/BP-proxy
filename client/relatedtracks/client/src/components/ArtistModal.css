.myModal {
    display: 'block'; /* Hidden by default */
    position: absolute; /* Stay in place */
    z-index: 1; /* Sit on top */
    left: 0px;
    top: 5px;
    margin: 10px;
    width: 130px; /* Full width */
    height: 200px; /* Full height */
    overflow: auto; /* Enable scroll if needed */
}

.modalContent {
    background-color: #ffffff;
    border: 1px solid;
    box-shadow: 1px 1px;
    border-radius: 5px;
    position: relative;
    width: 95%;
    height: 95%;
    text-align: center;
}

.rel_albumImgModal {
  display: block;
  margin: auto;
  width: 50px;
  height: 50px;
  border: 1px solid #999;
  border-radius: 50%;
  position: relative;
  top: 12px;
}

.rel_followerCountModal {
  font-size: 14px;
  display: inline-block; 
  text-align: center; 
}

.button {
  display: block;
  margin: auto;
}

.proIcon {
  color: #ff5500;
}

.rel_hover:hover {
  color: #000;
}







