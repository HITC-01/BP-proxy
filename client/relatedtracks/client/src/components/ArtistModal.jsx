import React from 'react';
import PropTypes from 'prop-types';
import styles from './ArtistModal.css';

const numeral = require('numeral');

const ArtistModal = (props) => {
  const { song } = props;
  const { mouseOut } = props;
  const toProfile = `Visit ${song.name}'s profile`;
  const followerCount = `${numeral(song.followers).format('0,0')} followers`;
  return (
    <div id={song.name} className={styles.myModal} onMouseLeave={() => { mouseOut(); }}>
      <div className={styles.modalContent}>
        <div>
          <img className={styles.rel_albumImgModal} src={song.album_img} alt="Album art" width="50" height="50" />
          <p title={toProfile} className={`${styles.rel_artistNameModal} ${styles.rel_hover}`}>
            {song.name}
            <i title="Pro Unlimited user" className={`fas fa-star ${styles.proIcon}`} />
          </p>
          <i className="fas fa-users" />
          <p title={followerCount} className={`${styles.rel_followerCountModal} ${styles.rel_hover}`}>{numeral(song.followers).format('0,0a')}</p>
          <input type="button" className={styles.button} value="Follow" />
        </div>
      </div>
    </div>
  );
};

ArtistModal.propTypes = {
  song: PropTypes.object.isRequired,
  mouseOut: PropTypes.func,
};

ArtistModal.defaultProps = {
  mouseOut: () => {},
};

export default ArtistModal;
