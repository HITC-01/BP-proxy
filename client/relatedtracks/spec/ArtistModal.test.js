import React from 'react';
import renderer from 'react-test-renderer';
import ArtistModal from '../client/src/components/ArtistModal';

describe('tests the ArtistModal component', () => {
  test('Should render correctly', () => {
    // mouseOut
    const songData = {
      title: 'Qui laboriosam official',
      name: 'Testing name',
      album_img: 'http://www.google.com',
      followers: 53534,
    };
    const component = renderer.create(<ArtistModal song={songData} />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
