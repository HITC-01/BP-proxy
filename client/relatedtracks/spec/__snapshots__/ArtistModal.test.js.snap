// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests the ArtistModal component Should render correctly 1`] = `
<div
  className="myModal"
  id="Testing name"
  onMouseLeave={[Function]}
>
  <div
    className="modal-content"
  >
    <div>
      <img
        alt="Album art"
        className="rel_albumImgModal"
        height="50"
        src="http://www.google.com"
        width="50"
      />
      <p
        className="rel_artistNameModal rel_hover"
        title="Visit Testing name's profile"
      >
        Testing name
        <i
          className="fas fa-star proIcon"
          title="Pro Unlimited user"
        />
      </p>
      <i
        className="fas fa-users"
      />
      <p
        className="rel_followerCountModal rel_hover"
        title="53,534 followers"
      >
        54k
      </p>
      <input
        className="button"
        type="button"
        value="Follow"
      />
    </div>
  </div>
</div>
`;
